src/client/src/Stream.c

```diff
@@ -3082,9 +3274,81 @@ STATUS appendValidatedMetadata(PKinesisVideoStream pKinesisVideoStream, PCHAR na
     // Set the persistent mode
     pSerializedMetadata->persistent = persistent;
 
+CleanUp:
+
+    // Free the allocation on error.
+    if (STATUS_FAILED(retStatus) && pSerializedMetadata != NULL) {
+        MEMFREE(pSerializedMetadata);
```

Comment:
This is OK but you can also remove the `pSerializedMetadata != NULL` check and just replace the call to MEMFREE with `SAFE_MEMFREE` which will do the null check.